pkg/process/v5/transformers/nvd/unique_pkg.go

```diff
@@ -63,14 +63,64 @@ func findUniquePkgs(cfgs ...nvd.Configuration) uniquePkgTracker {
 	return set
 }
 
-func determinePlatformCPEAndNodes(c nvd.Configuration) (*string, []nvd.Node) {
-	var platformCPE *string
+// nolint:gocognit
+func platformPackageCandidates(set uniquePkgTracker, c nvd.Configuration) bool {
+	nodes := c.Nodes
+	result := false
```

Comment:
once the below guard clause is adoped, this can be bumped closer to the loop where it is really used, making it a little more readible